### ISSUE TITLE
How to un-toggle key bind

### ISSUE THREAD

**Issue by tejedamikey** [virresh/matvt#58]
Okay so I have accidentally bound the “OK” button to my BENQ gv30 remote to the toggle mouse button and I can no longer select anything (only endlessly toggle) is there a key pattern I can push to reset the binding? Using a remote app doesn’t work either since it’s even recognized the app remote to have the same binding. There is also seemingly no way to use a physical mouse/keyboard to go in and undo my error.

**Comment by MailYouLater** [comment]
Well, the easiest answer is to move your selection to what you want, then push your center D-pad  button lots of times in quick succession hoping to overload the key interception and maybe, occasionally activate its default functionality, slowly working your way to where you can change a setting that affects this.

A more extreme answer is to performed a factory reset.  I'm unfamiliar with the BenQ GV30, but there should be reasonably simple instructions you can look up for factory reseting basically any Android (or Android TV) device.  Then you have to set everything up from scratch again.

Other options in situations similar to this include...
- If you can plug your Android TV device into a TV (or possibly audio receiver) that supports HDMI-CEC, try using its remote.  There's a chance that the OK, Enter or center D-pad button on it may not be intercepted by MATVT.
- If you have a BlueTooth mouse, keyboard or gamepad that's already paired to this device, try using it.  The left mouse button, Enter key, or whichever gamepad button(s) Android treats as activation button(s) shouldn't be intercepted by MATVT.
- Check if your device has a USB port, you may be able to plug in a mouse, keyboard or gamepad there.
- If it doesn't have a full size USB Type-A port, check to see if its power input is a USB Type-C or Micro-USB port.  If it's USB Type-C you should be able to use basically any powered USB Type-C dock, and if it's Micro-USB you'll need a powered USB-OTG adapter to add a USB Type-A port you can plug a mouse, keyboard or gamepad into.
- If you have Zank Android Remote set up on your Android TV device, then you can use that to control it, either with the trackpad style mouse input, the graphics tablet style (no visual feedback) touch input, or the remote access style functionality, where it screen captures your TV, displays that on your phone, and sends your touch inputs back to your Android TV device.

**Comment by boognish-rising** [comment]
I know this is an almost a month old, but in case you were looking for another solution or anybody else happens to experience a similar situation, (don't feel bad I had a similar experience when I set the boss key as the select button without really thinking about the ramifications) I found that while using the remote via the Google Home app, all the keys are mapped exactly as if you were actually pressing the key on the remote itself. However, the one key difference is that going with the virtual remote app via the Home app, you have the option to use the virtual on screen keyboard on your phone. You should still be able to use the arrow keys on the remote itself and then simply use your virtual screen based keyboard's Enter/Return key to toggle whatever you're looking to toggle that can't be due to this app's configuration. 

If you add a keyboard such as Unexpected Keyboard or Hacker's Keyboard (both on F-droid, Droid-ify, Aurora Store etc but might also be found here on GH) they feature arrow keys, Function keys and many (see screenshots below), actually giving you more key options than you would have with any standard 104 key keyboard (without constantly referencing the Alt + key combination and/or AltGr for keys not included as a physical key, anyway) 

I hope that helps you or anyone else that might have jammed themselves up by setting the boss key in a similar manner. It was certainly a revelation for me and helped me out of a very similar bind.

Happy navigating!

![Screenshot_20221227-072548.png](https://user-images.githubusercontent.com/76668060/209666584-dbe58065-de61-4a93-9216-c1cc093dc9f1.png)
![Screenshot_20221227-072403.png](https://user-images.githubusercontent.com/76668060/209666602-afa9c596-8e97-43ba-bb3c-3c80be6342e1.png)
![Screenshot_20221227-072333.png](https://user-images.githubusercontent.com/76668060/209666629-b2ae63a0-f576-488f-9850-b53da1bafb7d.png)
![Screenshot_20221227-072304.png](https://user-images.githubusercontent.com/76668060/209666655-56985c53-81fd-4c60-a56f-2eb8c727ce40.png)
![Screenshot_20221227-072237.png](https://user-images.githubusercontent.com/76668060/209666673-c71bf048-b999-47ef-8bfa-309c9220e284.png)
![Screenshot_20221227-072220.png](https://user-images.githubusercontent.com/76668060/209666708-c79fecc0-c61b-4237-8bdd-7415b5f5ba57.png)
![Screenshot_20221227-072213.png](https://user-images.githubusercontent.com/76668060/209666736-337cc642-e18b-4534-b0b9-59cdeca170ce.png)
![Screenshot_20221227-072204.png](https://user-images.githubusercontent.com/76668060/209666760-f7231284-5231-4b91-8a21-d9a0034f7b46.png)![Screenshot_20221227-072149.png](https://user-images.githubusercontent.com/76668060/209666787-2e0c4c15-facf-4d4b-95f2-da4a1feccac2.png)

**Comment by Ragaban** [comment]
Greetings, I have a Xiaomi TV Stick 4k. I did the same mistake o foverwritting my "Ok" button as the boss key and checking "Will boss key toggle". I did not set-up any other navigation devices with bluetooth. Could I connect a wired keyboard/mouse to the stick?

**Comment by Ragaban** [comment]
Ok here is a solution for anyone who did not install any software prior. Try to buy a "OTG Host Power Splitter Y Micro USB Male to USB Female Adapter Cable Cord" 
![image](https://github.com/virresh/matvt/assets/78890944/f9352b02-32c7-4d42-b6fb-99245cf00cd0)
Like this one in the image. I bought one for around 5 euros. With one of these you can power the Xiaomi mi stick and connect a keyboard to it. After connecting a keyboard you can navigate without the remote (with the arrow keys and the spacebar). You could connect a mouse but I found out that the left and right click do not work so a keyboard is more reliable.

**Comment by virresh** [comment]
Yep, any connector should be fine. Glad you found one that works for your case :) 

Alternatively, if you have a bluetooth mouse/keyboard with simultaneous multiple connections (these can be pretty expensive though, like the logitech mx720 mouse or k850 keyboard), it might be helpful to keep one of these paired and ready to use just in case.
